src/main/java/uk/gov/hmcts/reform/civil/service/citizen/HWFFeePaymentOutcomeService.java

```diff
@@ -25,36 +30,54 @@ public CaseData updateHwfReferenceNumber(CaseData caseData) {
                     .helpWithFee(YesOrNo.YES)
                     .helpWithFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome())
                     .build();
-                caseData.getCaseDataLiP().setHelpWithFees(helpWithFees);
+                updatedData.caseDataLiP(CaseDataLiP.builder().helpWithFees(helpWithFees).build());
+                helpWithFeesForTabService.setUpHelpWithFeeTab(updatedData);
             }
             if (caseData.isHWFTypeHearing()) {
-                caseData = caseData.toBuilder().hearingHelpFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome()).build();
+                updatedData.hearingHelpFeesReferenceNumber(caseData.getFeePaymentOutcomeDetails().getHwfNumberForFeePaymentOutcome()).build();
             }
-            clearHwfReferenceProperties(caseData);
+            clearHwfReferenceProperties(updatedData);
         }
-        return caseData;
+        return updatedData.build();
     }
 
-    private void clearHwfReferenceProperties(CaseData caseData) {
-        caseData.getFeePaymentOutcomeDetails().setHwfNumberAvailable(null);
-        caseData.getFeePaymentOutcomeDetails().setHwfNumberForFeePaymentOutcome(null);
+    private void clearHwfReferenceProperties(CaseData.CaseDataBuilder caseDataBuilder) {
+        CaseData caseData = caseDataBuilder.build();
+        caseDataBuilder.feePaymentOutcomeDetails(caseData.getFeePaymentOutcomeDetails().toBuilder().hwfNumberAvailable(null).build());
+        caseDataBuilder.feePaymentOutcomeDetails(caseData.getFeePaymentOutcomeDetails().toBuilder().hwfNumberForFeePaymentOutcome(null).build());
     }
 
-    public CaseData updateOutstandingFee(CaseData caseData) {
-
-        BigDecimal claimIssuedRemissionAmount = caseData.getClaimIssueRemissionAmount();
-        BigDecimal hearingRemissionAmount = caseData.getHearingRemissionAmount();
+    public CaseData updateOutstandingFee(CaseData caseData, HWFFeePaymentType paymentType) {
+        var updatedData = caseData.toBuilder();
+        BigDecimal claimIssuedRemissionAmount = HWFFeePaymentType.NO_REMISSION.equals(paymentType)
```

Comment:
Use == to compare enums.Check all places

Reply:
Updated from review